### PR TITLE
Support for Control Flow Guard

### PR DESCRIPTION
**Pull request by gkhanna79**
. This makes VS 2015 as the minimum required build toolset, on Windows, for the repo.

This fixes https://github.com/dotnet/coreclr/issues/3013.

@jkotas PTAL

### CODE REVIEW

**Comment by gkhanna79** [comment]
@dotnet-bot test this please

File: CMakeLists.txt [hunk 1]
```diff
@@ -430,6 +430,13 @@ if (WIN32)
   add_compile_options($<$<CONFIG:Debug>:/homeparams>) # Force parameters passed in registers to be written to the stack
   endif (CLR_CMAKE_PLATFORM_ARCH_AMD64)
 
+  if(NOT CLR_CMAKE_PLATFORM_ARCH_ARM64)
```
Comment (jkotas) [inline CMakeLists.txt]: For other products, we have been using AMD64 || I386 condition to enable it.

**Comment by jkotas** [comment]
LGTM modulo minor comment.

File: CMakeLists.txt (hunk 1 above)
Comment (gkhanna79) [inline CMakeLists.txt]: I am enabling it for all architectures on Windows aside from Arm64. Hence, the !ARM64.

**Comment by AndyAyersMS** [comment]
FYI, I am seeing frequent linker crashes now in x64 debug builds. They seem to be intermittent and go away if I rebuild. This is with update 1.
```
libcmtd.lib(loadcfg.obj) : fatal error LNK1000: Internal error during IMAGE::Pass2 [C:\Repos\coreclr\bin\obj\Windows_NT.x64.Debug\src\tools\crossgen\crossgen.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\x86_amd64\link.exe
Microsoft (R) Incremental Linker Version 14.00.23506.0
```

**Comment by gkhanna79** [comment]
Hmm - I didnt see those @AndyAyersMS  though I always did a clean build. Are there specific steps that lead you to repro it deterministically?

**Comment by AndyAyersMS** [comment]
Nothing special ... just edit a few .cpp files in the jit (or likely elsewhere) and `build x64 debug`, then repeat a few times. Probably happens about 1 time in 4 or so. Could be related to incremental linking I suppose.

**Comment by wtgodbe** [comment]
@gkhanna79 I'm no longer able to build with vs2013, despite having visual studio 2015 on my machine (i.e. './build.cmd vs2013' exits immediately with 'Visual Studio 2015+ is a prerequisite to build this repository') - is that an intended effect of this change? If so, why keep the option to build with vs2013?

**Comment by gkhanna79** [comment]
@wtgodbe Yes, this is intended and the instructions were updated for the same:

https://github.com/dotnet/coreclr/blob/master/Documentation/building/windows-instructions.md

I missed removing command line support for VS2013 and will remove it today.

**Pull request merged**